app/controllers/katello/api/v2/host_packages_controller.rb

```diff
@@ -20,10 +20,12 @@ class Api::V2::HostPackagesController < Api::V2::ApiController
 
     api :GET, "/hosts/:host_id/packages", N_("List packages installed on the host")
     param :host_id, :number, :required => true, :desc => N_("ID of the host")
+    param :packages_restrict_latest, :boolean, :desc => N_("Return only the latest version of each package")
     param_group :search, Api::V2::ApiController
     add_scoped_search_description_for(Katello::InstalledPackage)
     def index
       collection = scoped_search(index_relation, :name, :asc, :resource_class => ::Katello::InstalledPackage)
+      collection[:results] = HostPackagePresenter.with_latest(collection[:results], latest_upgradables) if ::Foreman::Cast.to_bool(params[:packages_restrict_latest])
```

Comment:
I'd recommend migrating this controller to work solely with your new presenter rather than two alternate paths (effectively).